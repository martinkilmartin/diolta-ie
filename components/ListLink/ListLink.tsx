import { Link } from '../Link'

type Props = {
  text: string
  href?: string
}

const ListLink = ({ text, href = '#' }: Props): JSX.Element => (
  <li className="my-2">
    <Link text={text} href={href} />
  </li>
)

export default ListLink
