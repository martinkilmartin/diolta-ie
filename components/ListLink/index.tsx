export { default as Link } from './ListLink'
