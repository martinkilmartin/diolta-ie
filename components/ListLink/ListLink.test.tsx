import renderer from 'react-test-renderer'
import ListLink from './ListLink'

test('ListLink exists with class', () => {
  const listLinkText = 'Click Me!'
  const listLinkRenderer = renderer.create(<ListLink text={listLinkText} />)
  const tree = listLinkRenderer.toJSON()
  expect(tree).toMatchSnapshot()
  const listLinkInstance = listLinkRenderer.root
  expect(
    listLinkInstance.findByProps({
      className: 'my-2',
    }).children.length
  ).toEqual(1)
  expect(
    listLinkInstance.findByProps({
      className: 'transition-colors duration-200 hover:text-neutral-focus',
    }).children[0]
  ).toEqual(listLinkText)
})
