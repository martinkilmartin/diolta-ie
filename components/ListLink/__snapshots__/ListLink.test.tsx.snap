// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListLink exists with class 1`] = `
<li
  className="my-2"
>
  <a
    className="transition-colors duration-200 hover:text-neutral-focus"
    href="#"
  >
    Click Me!
  </a>
</li>
`;
